### PR TITLE
Add missing_docs lint and missing docs

### PR DESCRIPTION
**Pull request by hug-dev**
Signed-off-by: Hugues de Valon <hugues.devalon@arm.com>

### CODE REVIEW

File: src/bin/main.rs [hunk 1]
```diff
@@ -1,5 +1,14 @@
...
 // SPDX-License-Identifier: Apache-2.0
+//! Parsec is the Platform AbstRaction for SECurity, a new open-source initiative to provide a
+//! common API to secure services in a platform-agnostic way.
+//!
+//! Parsec documentation is available at:
+//! https://parallaxsecond.github.io/parsec-book/index.html
+//!
+//! Most of Parsec configuration comes from its configuration file.
+//! Please check the documentation to find more about configuration:
+//! https://parallaxsecond.github.io/parsec-book/user_guides/configuration.html
```
Comment (ionut-arm) [inline src/bin/main.rs:11]: I think you can use these like markdown links:

```suggestion
//! Please check the documentation to find more about configuration
//! [here](https://parallaxsecond.github.io/parsec-book/user_guides/configuration.html)
```

File: src/providers/mod.rs [hunk 2]
```diff
@@ -22,32 +22,45 @@ pub mod mbed_provider;
 #[cfg(feature = "tpm-provider")]
 pub mod tpm_provider;
 
+/// Provider configuration structure
+/// For providers configs in parsec config.toml we use a format similar
```
Comment (ionut-arm) [inline src/providers/mod.rs:26]: `Parsec` with capital P? Or was this intended?

File: src/front/listener.rs [hunk 3]
```diff
@@ -9,20 +9,25 @@ use derivative::Derivative;
...
+    /// Listener using Unix Domain Socket
     DomainSocket,
 }
 
+/// Configuration of the Listener
 #[derive(Copy, Clone, Deserialize, Debug)]
 pub struct ListenerConfig {
+    /// Type of the Listener
     pub listener_type: ListenerType,
+    /// Timeout of the Listener before the connection errors out
```
Comment (ionut-arm) [inline src/front/listener.rs:30]: In seconds? Milliseconds?

**Review by ionut-arm — COMMENTED** [review]
Thanks! This is great! I noticed that some stuff was missing docs, didn't know how come clippy wasn't complaining.

**Comment by hug-dev** [comment]
Applied all comments!

**Review by ionut-arm — APPROVED** [review]
🚀 📜

**Pull request merged**